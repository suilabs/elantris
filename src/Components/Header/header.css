.sui-page-header {
  width: 100%;
  text-align: center;
  margin-top: 1rem; }
  .sui-page-header .sui-logo {
    position: relative;
    left: 50%;
    transform: translateX(-50%); }
