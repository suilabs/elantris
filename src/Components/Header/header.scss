@import '../../base.scss';

.sui-page-header {
  position: static;
  width: 100%;
  margin: $sui-spacing-large 0;

  &__wrapper {
    position: relative;
    width: $sui-page-max-width;

    @include h-centered;
  }

  .sui-logo {
    float: left;
  }

  .sui-navbar-wrapper {
    position: absolute;
    right: 0;
    bottom: 0;
    width: 70%;
    border-bottom: $sui-pixel $sui-color-primary dotted;
    text-align: right;
  }
}
