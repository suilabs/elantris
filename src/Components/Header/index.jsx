import Component from './header';

export default Component;
