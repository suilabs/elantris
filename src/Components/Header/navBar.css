.sui-navbar {
  padding-top: 1rem;
  border-bottom: 1px dashed mediumvioletred;
  left: 50%;
  display: inline-block; }
  .sui-navbar ul {
    list-style-type: none;
    margin: 0;
    padding: 0; }
  .sui-navbar__item {
    transition: background-color .1s;
    display: inline; }
    .sui-navbar__item:first-child {
      padding-left: 1rem; }
    .sui-navbar__item:last-child {
      padding-right: 1rem; }
    .sui-navbar__item:hover {
      background-color: rgba(0, 0, 0, 0.2); }
    .sui-navbar__item__link {
      text-decoration: none; }
      .sui-navbar__item__link:visited {
        color: initial; }
  .sui-navbar__item-spacer {
    padding: 0 0.25rem;
    display: inline; }
