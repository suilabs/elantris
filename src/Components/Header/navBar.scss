.sui-navbar {
  padding-top: 1rem;
  border-bottom: 1px dashed mediumvioletred;
  left: 50%;
  display: inline-block;
  //transform: translateX(-50%);

  ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
  }

  &__item {
    transition: background-color .1s;
    display: inline;

    &:first-child {
      padding-left: 1rem;
    }

    &:last-child {
      padding-right: 1rem;
    }

    &:hover {
      background-color: rgba(0,0,0,0.2);
    }

    &__link {
      text-decoration: none;

      &:visited {
        color: initial;
      }

    }
  }

  &__item-spacer {
    padding: 0 0.25rem;
    display: inline;
  }
}