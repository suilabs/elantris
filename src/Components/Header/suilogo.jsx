import React from 'react';
import PropTypes from 'proptypes';

import './suilogo.css';

const SuiLogo = (props) => {
  const { width, height } = props;
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      id="svg3109"
      viewBox="0 0 480 250"
      width={width}
      height={height}
      version="1.1"
    >
      <g
        id="layer1"
        transform="translate(-186.70713,-512.31933)"
      >
        <path
          id="path3055"
          d="m 205.50713,707.91933 c 7.5,14 20.1,27.3 44.3,27.3 17.7,0 32.7,-10.2 32.7,-28.3 0,-40.9 -88.9,-26.9 -88.9,-86.9 0,-30 24.9,-48 56.9,-48 33.1,0 50.1,20.8 54.9,34.1 l -19.9,11.2 c -4.8,-9.5 -14,-22.8 -36.8,-22.8 -16.4,0 -29.3,9.2 -29.3,24.5 0,39.9 89.3,24.9 89.3,86.6 0,31.3 -25.9,52.5 -60.3,52.5 -37.1,0 -53.8,-21.1 -61.7,-37.8 l 18.8,-12.4 z"
        />
        <path
          id="path3057"
          d="m 340.40713,687.21933 v -110.8 h 26.9 v 111.8 c 0,28.3 18.1,45.3 41.9,45.3 27.3,0 46,-20.1 46,-49.4 v -107.7 h 27.3 v 177.2 h -26.9 v -25.6 c -8.2,16 -28.3,30 -52.5,30 -41.2,0 -62.7,-30.6 -62.7,-70.8 z"
        />
        <path
          id="path3059"
          d="m 540.40713,512.41933 c 10.6,0 18.7,8.2 18.7,18.4 0,10.6 -8.2,18.7 -18.7,18.7 -10.2,0 -18.4,-8.2 -18.4,-18.7 0,-10.3 8.2,-18.4 18.4,-18.4 z m -13.3,64 h 26.9 v 177.2 h -26.9 v -177.2 z"
        />
        <path
          style={{ fill: '#881354' }}
          id="path3061"
          d="m 577.10713,692.01933 v -8.6 h 78.4 v 8.6 h -78.4 z"
        />
        <path
          style={{ fill: '#881354' }}
          id="path3063"
          d="m 619.60713,650.71933 c 0,-7.2 2.7,-12.6 5.1,-14.9 h -6.9 c -7.9,0 -12.5,5.3 -12.5,13.7 0,6.9 2.5,12.6 4.9,16.6 l -6.3,3.5 c -2.4,-3.8 -6.2,-11.3 -6.2,-20.8 0,-15.4 8.6,-21.5 20.6,-21.5 h 37.2 v 7.4 l -4.8,1 c 3.4,2.9 6.2,7.7 6.2,15.3 0,12.4 -7.8,21.5 -18.8,21.5 -10.7,-0.1 -18.5,-9.1 -18.5,-21.8 z m 30.2,-1.2 c 0,-8.2 -4.9,-14 -11.7,-14 -7,0 -11.6,5.6 -11.6,14.1 0,8.6 5,14.1 11.6,14.1 6.4,0 11.7,-5.5 11.7,-14.2 z"
        />
        <path
          style={{ fill: '#881354' }}
          id="path3065"
          d="m 577.10713,613.41933 v -8.6 h 31.1 c -2.9,-1.7 -10.5,-8.1 -10.5,-19.9 0,-15.9 12.4,-26.9 29.7,-26.9 17.2,0 29.5,11.5 29.5,27.5 0,12.3 -8.2,18.2 -10.3,19.6 h 8.9 v 8.2 h -78.4 z m 72.1,-27.3 c 0,-10.8 -8.6,-19.3 -21.7,-19.3 -12.9,0 -22,8 -22,19.1 0,10.6 8,19.3 22,19.3 11.8,0 21.7,-7.4 21.7,-19.1 z"
        />
        <path
          style={{ fill: '#881354' }}
          id="path3067"
          d="m 641.00713,545.31933 c 4.4,-2.4 8.7,-6.4 8.7,-14.1 0,-5.6 -3.3,-10.4 -9,-10.4 -13,0 -8.6,28.3 -27.6,28.3 -9.5,0 -15.3,-7.9 -15.3,-18.1 0,-10.5 6.6,-15.9 10.8,-17.5 l 3.6,6.3 c -3,1.5 -7.3,4.4 -7.3,11.7 0,5.2 2.9,9.3 7.8,9.3 12.7,0 7.9,-28.4 27.5,-28.4 10,0 16.7,8.2 16.7,19.2 0,11.8 -6.7,17.1 -12,19.6 l -3.9,-5.9 z"
        />
        <circle
          d="m 831.50002,526.79999 c 0,10.27249 -8.3275,18.6 -18.6,18.6 -10.27249,0 -18.6,-8.32751 -18.6,-18.6 0,-10.2725 8.32751,-18.6 18.6,-18.6 10.2725,0 18.6,8.3275 18.6,18.6 z"
          id="circle3069"
          r="18.6"
          cy="526.79999"
          cx="812.90002"
          className="sui-logo__vertical-dot"
          transform="translate(-172.49287,212.41933)"
        />
        <circle
          d="m 731.69998,318.5 c 0,10.2725 -8.32751,18.6 -18.6,18.6 -10.2725,0 -18.6,-8.3275 -18.6,-18.6 0,-10.2725 8.3275,-18.6 18.6,-18.6 10.27249,0 18.6,8.3275 18.6,18.6 z"
          id="circle3071"
          r="18.6"
          cy="318.5"
          cx="713.09998"
          className="sui-logo__horizontal-dot"
          transform="translate(-172.49287,212.41933)"
        />
      </g>
    </svg>);
};

SuiLogo.propTypes = {
  width: PropTypes.string,
  height: PropTypes.string,
};

SuiLogo.defaultProps = {
  width: null,
  height: null,
};

export default SuiLogo;
