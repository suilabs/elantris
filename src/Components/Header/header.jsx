import React from 'react';
import {Link} from 'react-router-dom';

import NavBar from './navBar';

import SuiLogo from './suilogo';

import './header.css';

const Header = () => (
  <div className="sui-page-header">
    <div className="sui-logo">
      <Link to="/">
        <SuiLogo width="100px" />
      </Link>
    </div>
    <NavBar />
  </div>
);

export default Header;
