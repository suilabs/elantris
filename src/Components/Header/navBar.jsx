import React from 'react';
import { withRouter, Link } from 'react-router-dom';
import PropTypes from 'proptypes';

import './navBar.css';

const NavBarItem = (props) => {
  const root = props.history.location.pathname.split('/')[1];
  const active = `/${root}` === props.to ? '--active' : '';
  const classname = `sui-navbar__item${active} menu-item`;
  return (
    <li className={classname}>
      <Link id={props.to} to={props.to} className="sui-navbar__item--link">
        {props.label}
      </Link>
    </li>
  );
};

NavBarItem.propTypes = {
  to: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  history: PropTypes.shape({
    location: PropTypes.shape({
      pathname: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
};

const NavBarItemWR = withRouter(NavBarItem);

export const NavBarSpacer = () => (
  <li className="sui-navbar__item-spacer">
    Â·
  </li>
);

const NavBar = () => (
  <nav className="sui-navbar">
    <ul>
      <NavBarItemWR to="/design" label="Design" />
      <NavBarItemWR to="/software" label="Software" />
      <NavBarItemWR to="/about" label="About us" />
    </ul>
  </nav>
);

export default NavBar;
