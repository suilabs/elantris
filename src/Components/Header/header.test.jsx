import React from 'react';

import Header from './header';

describe('Header', () => {
  it('renders correctly', () => {
    const tree = shallow(
      <Header/>
    );
    
    expect(tree).toMatchSnapshot();
  });
});
