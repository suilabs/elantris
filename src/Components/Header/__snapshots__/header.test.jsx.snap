// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  "toJson": [Function],
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="sui-page-header__wrapper clearfix"
>
        <div
                className="sui-logo"
        >
                <Link
                        replace={false}
                        to="/"
                >
                        <SuiLogo
                                height={null}
                                width="100px"
                        />
                </Link>
        </div>
        <div
                className="sui-navbar-wrapper"
        >
                <NavBar />
        </div>
</div>,
      "className": "sui-page-header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="sui-logo"
>
            <Link
                        replace={false}
                        to="/"
            >
                        <SuiLogo
                                    height={null}
                                    width="100px"
                        />
            </Link>
</div>,
          <div
            className="sui-navbar-wrapper"
>
            <NavBar />
</div>,
        ],
        "className": "sui-page-header__wrapper clearfix",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/"
>
              <SuiLogo
                            height={null}
                            width="100px"
              />
</Link>,
            "className": "sui-logo",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <SuiLogo
                height={null}
                width="100px"
/>,
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "height": null,
                "width": "100px",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavBar />,
            "className": "sui-navbar-wrapper",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="sui-page-header__wrapper clearfix"
>
          <div
                    className="sui-logo"
          >
                    <Link
                              replace={false}
                              to="/"
                    >
                              <SuiLogo
                                        height={null}
                                        width="100px"
                              />
                    </Link>
          </div>
          <div
                    className="sui-navbar-wrapper"
          >
                    <NavBar />
          </div>
</div>,
        "className": "sui-page-header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="sui-logo"
>
              <Link
                            replace={false}
                            to="/"
              >
                            <SuiLogo
                                          height={null}
                                          width="100px"
                            />
              </Link>
</div>,
            <div
              className="sui-navbar-wrapper"
>
              <NavBar />
</div>,
          ],
          "className": "sui-page-header__wrapper clearfix",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/"
>
                <SuiLogo
                                height={null}
                                width="100px"
                />
</Link>,
              "className": "sui-logo",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <SuiLogo
                  height={null}
                  width="100px"
/>,
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "height": null,
                  "width": "100px",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavBar />,
              "className": "sui-navbar-wrapper",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
